### DISCUSSION TITLE
Auswahlliste aus gViewService

### DISCUSSION THREAD

**Discussion by marko-jer** [e-netze/webgis-community#210] [General]
Hallo Jürgen,
wenn ich für ein Suchfeld  zB "Gemeindename" die Auswahlliste im CMS aus dem gView-Service (#) generiere, dann beinhaltet die Auswahlliste exakt 250 "Gemeindenamen". Die restlichen Gemeindenamen werden in der Auswahlliste nicht mehr angezeigt, auch wenn ich fast den ganzen Gemeindenamen hineinschreibe. (zB Voitsberg) - kein Nachladen.
Kann man diesen Wert (250) irgendwo ändern?

Ausprobieren kannst es in unserer Basiskarte - Dienst Orientierung (gView-Service).

lg m

**Comment by jugstalt** [comment]
Kann ich nicht nachvollziehen?! Im Steiermark Atlas (Orientierung/Gemeindesuche) schaut das für mich ganz normal aus. Besteht dieses Problem noch?

**Comment by marko-jer** [comment]
Ja, besteht nach wie vor.
SQL-Abfrage der Gemeinden sortiert nach Gemeindenamen (links die laufe Nummer des Datensatzes)
![image](https://github.com/user-attachments/assets/886adf0b-bb2c-4f0a-bc1d-2cd833ffbb14)

und im folgenden Screenshot die Auswahlliste, wenn man "stra" eingibt, Es wird nur der 250. Datensatz "Straden" noch aufgelistet.
![image](https://github.com/user-attachments/assets/17944ad6-2682-4574-ba6a-d9f0ce86d33a)

Wie im ersten Bild sichtbar, sollten da mindestens noch "Stralleg" und "Straß in der Steiermark" aufgelistet werden.

**Comment by gis-eni** [comment]
Ok, jetzt versteh ich glaub ich das Problem. Du hast bei beim ConnectionString ``#`` eingetragen => frag den Dienst ab. Das passt so weit.
WebGIS macht dann ein Distinct auf den Dienst und holt die ersten 250 Elemente und gibt die passenden Element in der List aus (siehe oben)

**Achtung:** Mit ``#`` ersparst du dir nur den ConnectionString auf die Datenbank. Wie abgefragt wird musst dem WebGIS schon auch noch sagen. Auswahllisten können kaskadierend sein, willst du mit Wildcards vorne und/oder hintern suchen etc.
Wenn du keine Where Klausel angibst, werden einfach immer die ersten 250 Ergebnisse gesucht und dann noch einmal auf deine Eingabe gefiltert... 

Richtig wäre hier:

![image](https://github.com/user-attachments/assets/3048fa40-dd01-4733-971b-f6167c6ae639)

Dann übergibt WebGIS auch den eingebeben Wert und den Feature Server!

**Fazit:** nur ``#`` angeben, ohne Query sollte du bei kleinen Datensätzen (<250). Eigentlich sollst du immer eine Query angeben, ist definitiv effizienter als jedesmal alles abzuholen.

**Reply by gis-eni** [reply]
PS: [such1] ist hier der Platzhalter- ``such1`` ist hier der url-Name für das Suchfeld.

**Reply by marko-jer** [reply]
Danke! So funktioniert es 👍
